.github/workflows/ci.yml

```diff
@@ -0,0 +1,101 @@
+name: CI
+
+on:
+  push:
+    branches:
+      - main
+      - develop
+
+    tags:
+      - '*'
+
+  pull_request:
+
+env:
+  MAIN_REPO: Oceans-1876/challenger-portal
+
+jobs:
+
+# ----------------------------------------------------------------------
+  # NODE BUILD
+  # ----------------------------------------------------------------------
+  build:
+    runs-on: ubuntu-latest
+
+    strategy:
+      fail-fast: false
+      matrix:
+        node:
+          - "12"
+          - "14"
+          - "16"
+    steps:
+    # checkout source code
+    - uses: actions/checkout@v2
+
+    # setup node
+    - uses: actions/setup-node@v2
+      with:
+        node-version: ${{ matrix.node }}
+
+    # install dependencies
+    - name: install dependencies
+      run: npm install
+
+    # compile UI
+    - name: build
+      run: npm run build
+
+    # ----------------------------------------------------------------------
```

Comment:
We can skip the whole docker part. Considering we only need the final static files, I think we can get them from the first job and later upload them to our servers. I remember seeing some actions for uploading/downloading assets. We can probably use those as the last step of the `build` job.

Reply:
I'm not sure what you mean? You don't want to push separate versioned frontend docker images for incremental releases but rather just copy the static js/etc to deploy directly somewhere? What I like about the above is we could have one small VM with a docker-compose referring to our frontend:develop (and the API) and check every so often for a new image and pull it for auto deployment/demo purpose.

Reply:
The idea is to keep things as simple as possible and avoid adding complexity. Using docker for this code means adding an extra layer between the reverse proxy and the bundle.

Considering the UI app will have a continuous dev and deploy process, as it's an app that's going to be used and deployed by us, keeping pre-built docker images won't give us much and creates another dependency to think about, i.e. the docker image registry. We can always use git tags to specify which build we want to use.